# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1544670122 18000
#      Wed Dec 12 22:02:02 2018 -0500
# Node ID 39e84bf92d1801d2772edb8b721c8e6e9dfed3d8
# Parent  c0ac6fc191d7074c2e728164fc16802300553870
accept functions in nested +pkg directories in @DOCSTRING macros (bug #52596)

* scripts/mk-doc.pl: Detect nested +pkg directory names.

diff --git a/scripts/mk-doc.pl b/scripts/mk-doc.pl
--- a/scripts/mk-doc.pl
+++ b/scripts/mk-doc.pl
@@ -52,18 +52,25 @@ MFILE: foreach my $m_fname (@ARGV)
   my $fcn;
   if ($paths[-2] =~ m/^@/)
     {
       ## @classes will have @class/method as their function name
       $fcn = File::Spec->catfile (@paths[-2, -1]);
     }
   elsif ($paths[-2] =~ m/^\+/)
     {
-      ## +package functions have package.name their function name
-      $fcn = substr ($paths[-2], 1) . "." . $paths[-1];
+      $fcn = $paths[-1];
+      for (my $i = 2; $i < @paths; $i++)
+        {
+          if ($paths[-$i] =~ m/^\+/)
+            {
+              ## +package functions have package.name their function name
+              $fcn = substr ($paths[-$i], 1) . "." . $fcn;
+            }
+        }
     }
   else
     {
       $fcn = $paths[-1];
     }
 
   my @help_txt = gethelp ($fcn, $full_fname);
   next MFILE unless @help_txt;
